### PR TITLE
Feat/mongodb

### PR DESCRIPTION
**Pull request by xiangtch**
脚手架支持 mongodb

### CODE REVIEW

**Comment by JackyHweng** [comment]
# [Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=h1) Report
> Merging [#17](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=desc) into [master](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/commit/db297a5a42f3775eefe9af48c52c7008eac483f0&el=desc) will **increase** coverage by `0.11%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #17      +/-   ##
==========================================
+ Coverage   96.13%   96.25%   +0.11%     
==========================================
  Files          24       25       +1     
  Lines         259      267       +8     
==========================================
+ Hits          249      257       +8     
  Misses         10       10              
```


| [Impacted Files](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...ators/app/handler/configurer/mongodb\_configurer.js](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17/diff?src=pr&el=tree#diff-Z2VuZXJhdG9ycy9hcHAvaGFuZGxlci9jb25maWd1cmVyL21vbmdvZGJfY29uZmlndXJlci5qcw==) | `100.00% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=footer). Last update [db297a5...5463e14](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: generators/app/templates/${artifactId}-provider/src/main/java/${basePath}/controller/if_demo_mongodb.MongodbDemoController.tmpl.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("demo/mongodbs")
+@Payload
+public class MongodbDemoController {
```
Comment (taccisum) [inline generators/app/templates/${artifactId}-provider/src/main/java/${basePath}/controller/if_demo_mongodb.MongodbDemoController.tmpl.java]: 补充 `create` `update` `delete` 的方法

File: generators/app/templates/${artifactId}-provider/src/main/resources/db/sql/if_demo_mongodb.documentDemo_data.tmpl.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+use devops
+db.user.save({username:'demo', password: 'demo'})
```
Comment (taccisum) [inline generators/app/templates/${artifactId}-provider/src/main/resources/db/sql/if_demo_mongodb.documentDemo_data.tmpl.sql]: 我刚本地测试时，这些 demo 数据似乎没有生效？

Comment (taccisum) [reply generators/app/templates/${artifactId}-provider/src/main/resources/db/sql/if_demo_mongodb.documentDemo_data.tmpl.sql]: 能否做到启动时自动在 mongo 的库里执行这些语句？

Comment (taccisum) [reply generators/app/templates/${artifactId}-provider/src/main/resources/db/sql/if_demo_mongodb.documentDemo_data.tmpl.sql]: 另外就是，文件不应该命名为 sql 吧？

**Comment by JackyHweng** [comment]
# [Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=h1) Report
> Merging [#17](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=desc) into [master](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/commit/6044da20fbbbb0ef1690713dfd96b6b733729849&el=desc) will **increase** coverage by `0.11%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #17      +/-   ##
==========================================
+ Coverage   96.22%   96.33%   +0.11%     
==========================================
  Files          24       25       +1     
  Lines         265      273       +8     
==========================================
+ Hits          255      263       +8     
  Misses         10       10              
```


| [Impacted Files](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...ators/app/handler/configurer/mongodb\_configurer.js](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17/diff?src=pr&el=tree#diff-Z2VuZXJhdG9ycy9hcHAvaGFuZGxlci9jb25maWd1cmVyL21vbmdvZGJfY29uZmlndXJlci5qcw==) | `100.00% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=footer). Last update [6044da2...56ad935](https://codecov.io/gh/deepexi/generator-deepexi-spring-cloud/pull/17?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**